src/server/frontend_wayland/wl_client.cpp

```diff
@@ -38,10 +38,14 @@ struct ConstructionCtx
     ConstructionCtx(
         std::shared_ptr<msh::Shell> const& shell,
         std::shared_ptr<mf::SessionAuthorizer> const& session_authorizer,
-        std::function<void(mf::WlClient&)>&& client_created_callback)
+        std::function<void(mf::WlClient&)> const& client_created_callback,
+        std::function<bool(std::shared_ptr<ms::Session> const&, char const*)> const& extension_filter)
         : shell{shell},
           session_authorizer{session_authorizer},
-          client_created_callback{std::make_unique<std::function<void(mf::WlClient&)>>(std::move(client_created_callback))}
+          client_created_callback{std::make_unique<std::function<void(mf::WlClient&)>>(
+              client_created_callback)},
```

Comment:
It's a small pessimisation, but why are we now copying `client_created_callback` instead of moving it? The callback created in `WaylandConnector::WaylandConnector()` is not used elsewhere.

Reply:
Changed to move()